catalog/utilities/media_props_gen/generate_media_properties.py

```diff
@@ -0,0 +1,147 @@
+"""Automatic media properties generation."""
+
+import logging
+from dataclasses import dataclass
+from pathlib import Path
+
+from common.constants import MEDIA_TYPES
+from utilities.media_props_gen.helpers.column_parser import parse_python_columns
+from utilities.media_props_gen.helpers.db import FieldSqlInfo, create_db_props_dict
+from utilities.media_props_gen.helpers.md import Md
+
+
+log = logging.getLogger(__name__)
+# Silence noisy modules
+logging.getLogger("common.storage.media").setLevel(logging.WARNING)
+
+# Constants
+DOCS = Path(__file__).parent / "docs"
+DOC_MD_PATH = Path(__file__).parent / "media_properties.md"
+SOURCE_MD_PATH = DOCS / "media_props.md"
+
+PREAMBLE = open(DOCS / "preamble.md").read()
+POSTAMBLE = open(DOCS / "postamble.md").read()
+
+
+@dataclass
+class FieldInfo:
+    name: str
+    nullable: bool
+    datatype: str
+    constraint: str
```

Comment:
You're right. I was trying to explicitly state everywhere whether a value is nullable or not to prevent the errors when we have an unexpected `NULL` in a field upstream. Do you think it's better to convert it to a part of a constraint? I will open an issue for it if you think so, @dhruvkb

Reply:
It's okay as it is, makes the nullability easier to read at a glance. I'm just noting a difference between the this and the [API docs](https://docs.openverse.org/meta/media_properties/api.html) which put `not null` in the constraints column.